### PR TITLE
Kmesh: Add waypoint support for connect and deliver encoder dst info to sendmsg ebpf

### PR DESCRIPTION
**Pull request by bitcoffeeiux**
Ebpf connect4 program support connect to waypoint if the backend config waypoint ip and port.

Waypoint need dst info to routing, now connect4 can deliver these info to sendmsg ebpf, sendmsg can encode dst info by TLV.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: bpf/kmesh/workload/sockops_tuple.c [hunk 1]
```diff
@@ -49,7 +50,7 @@ static inline void build_auth_key(struct bpf_sock_ops *skops, struct bpf_sock_tu
 }
 
 // clean map_of_auth
-static inline void clean_auth_map(struct bpf_sock_ops *skops)
```
Comment (hzxuzhonghu) [inline bpf/kmesh/workload/sockops_tuple.c:52]: why change it

File: bpf/kmesh/workload/sockops_tuple.c [hunk 2]
```diff
@@ -59,6 +60,14 @@ static inline void clean_auth_map(struct bpf_sock_ops *skops)
     }
 }
 
+static inline void clean_send_map(struct bpf_sock_ops *skops)
```
Comment (hzxuzhonghu) [inline bpf/kmesh/workload/sockops_tuple.c:63]: seems this should be `clean_dstinfo_map`

File: bpf/kmesh/workload/sockops_tuple.c (hunk 1 above)
Comment (bitcoffeeiux) [reply bpf/kmesh/workload/sockops_tuple.c:52]: Sorry, clean_auth_map is modified by mistake when the function name is modified. As described below, clean_send_map= > clean_dstinfo_map should be modified.

File: bpf/kmesh/workload/sockops_tuple.c [hunk 3]
```diff
@@ -59,6 +60,14 @@ static inline void clean_auth_map(struct bpf_sock_ops *skops)
     }
 }
 
+static inline void clean_dstinfo_map(struct bpf_sock_ops *skops)
+{
+    __u32 *key = (__u32 *)skops->sk;
+    long ret = bpf_map_delete_elem(&map_of_dst_info, &key);
+    if(ret && ret != -ENOENT)
+        BPF_LOG(INFO, SOCKOPS, "map_of_dst_info bpf_map_delete_elem failed, ret: %d\n", ret);
```
Comment (LiZhenCheng9527) [inline bpf/kmesh/workload/sockops_tuple.c:68]:
```suggestion
        BPF_LOG(INFO, SOCKOPS, "bpf map delete destination info failed, ret: %d\n", ret);
```

File: bpf/kmesh/workload/include/backend.h [hunk 4]
```diff
@@ -28,7 +32,31 @@ static inline backend_value *map_lookup_backend(const backend_key *key)
...
 static inline int backend_manager(ctx_buff_t *ctx, backend_value *backend_v)
 {
+	int ret;
 	address_t target_addr;
+	__u32 *sk = (__u32 *)ctx->sk;
+	struct bpf_sock_tuple value_tuple= {0};
+
+	if (backend_v->waypoint_addr != 0 && backend_v->waypoint_port != 0) {
+		BPF_LOG(DEBUG, BACKEND, "find waypoint addr=[%u:%u]\n", backend_v->waypoint_addr,
+			backend_v->waypoint_port);
```
Comment (YaoZengzeng) [inline bpf/kmesh/workload/include/backend.h:42]: In ambient, waypoint listens on 15008 by default. But in kmesh, waypoint should listen on 15019, maybe we could hardcode it in this release and make it configurable in the future.

File: bpf/kmesh/workload/include/workload_common.h [hunk 5]
```diff
@@ -36,6 +36,8 @@
 #define CGROUP_SOCK_ERR		0
 #define CGROUP_SOCK_OK		1
 
+#define KMESH_TAIL_CALL_FILTER_CHAIN 0
```
Comment (supercharge-xsy) [inline bpf/kmesh/workload/include/workload_common.h:39]: What's this for?

File: bpf/kmesh/workload/include/backend.h (hunk 4 above)
Comment (bitcoffeeiux) [reply bpf/kmesh/workload/include/backend.h:42]: We currently hardcode it in the  kmesh-daemon code.

File: bpf/kmesh/workload/include/workload_common.h (hunk 5 above)
Comment (bitcoffeeiux) [reply bpf/kmesh/workload/include/workload_common.h]: Redundant code, deleted

File: bpf/kmesh/workload/include/backend.h (hunk 4 above)
Comment (bitcoffeeiux) [reply bpf/kmesh/workload/include/backend.h:42]: this PR: https://github.com/kmesh-net/kmesh/pull/220/files

File: bpf/kmesh/workload/cgroup_sock.c [hunk 6]
```diff
@@ -50,18 +51,41 @@ static inline int sock4_traffic_control(struct bpf_sock_addr *ctx)
...
 	BPF_LOG(DEBUG, KMESH, "origin addr=[%u:%u]\n", ctx->user_ip4, ctx->user_port);
 	frontend_v = map_lookup_frontend(&address);
-	if (frontend_v == NULL) {
-		BPF_LOG(ERR, KMESH, "find frontend failed\n");
-		return -ENOENT;
+	if (!frontend_v) {
+		address.service_port = 0;
+		frontend_v = map_lookup_frontend(&address);
+		if (!frontend_v) {
+			BPF_LOG(ERR, KMESH, "find frontend failed\n");
```
Comment (hzxuzhonghu) [inline bpf/kmesh/workload/cgroup_sock.c:58]: add the address in the log

Comment (hzxuzhonghu) [reply bpf/kmesh/workload/cgroup_sock.c:58]: And not err, this can happen commonly if the dst address is not in the mesh

**Pull request merged**